gmail.js

```diff
@@ -104,172 +105,144 @@ async function getEmailStats(user, emails) {
 
 /**
  * Determines whether email domain is managed by Google.
- * 1) Extract domain name
- * 2) Execute shell child process to check host (@source: https://stackabuse.com/executing-shell-commands-with-node-js/)
+ * @source for email-verifier: https://www.npmjs.com/package/email-verifier.
  * @param {string} emailAddress
  *
  * ex) team@calblueprint.org => true
  * reject(`error: ${error.message}`)
  * reject(`stderr: ${stderr}`)
+ *
+ * note: email-verifier 1) has no Promise support and 2) a rate limit of 10/second
  */
 function isGmailHosted(emailAddress) {
-  const domain = emailAddress.match(/(?<=@)[^.]+(.(\w+))*/g);
+  const verifier = new Verifier(process.env.AWS_EMAIL_VERIFICATION_KEY);
   return new Promise((resolve) => {
-    if (domain.length) {
-      exec(`host ${domain[0]}`, (error, stdout, stderr) => {
-        if (error) resolve(false);
-        if (stderr) resolve(false);
-        resolve(stdout.includes("google.com"));
-      });
-    }
+    verifier.verify(emailAddress, (err, data) => {
+      if (err) resolve(false);
+      resolve(
+        data &&
+          data.mxRecords &&
+          data.mxRecords.length > 0 &&
+          data.mxRecords[0].includes("google.com")
+      );
+    });
   });
 }
 
 /**
- * Create an OAuth2 client with the given credentials, and then execute the
- * given callback function.
+ * Create an OAuth2 client with the given credentials.
  * @param {Object} credentials The authorization client credentials.
- * @param {function} callback The callback to call with the authorized client.
+ * @returns {google.auth.OAuth2} The OAuth2 client to run subsequent calls to Gmail API.
  */
-function authorize(credentials, callback) {
+async function authorize(credentials) {
   const { client_secret, client_id, redirect_uris } = credentials.installed;
   const oAuth2Client = new google.auth.OAuth2(
     client_id,
     client_secret,
     redirect_uris[0]
   );
 
-  // Check if we have previously stored a token.
-  fs.readFile(TOKEN_PATH, (err, token) => {
-    if (err) return getNewToken(oAuth2Client, callback);
-    oAuth2Client.setCredentials(JSON.parse(token));
-    callback(oAuth2Client);
-  });
+  let token;
+  try {
+    token = await fs.readFile(TOKEN_PATH);
```

Comment:
small nit: you can technically do the JSON.parse here so we end up with the actual token rather than the tokenString